src/wasm/wasm-s-parser.cpp

```diff
@@ -1062,13 +1062,24 @@ Expression* SExpressionWasmBuilder::makeBlock(Element& s) {
     }
     curr->name = nameMapper.pushLabelName(sName);
     if (i >= s.size()) break; // empty block
+
+    // block signature
+    // TODO(sbc): Remove support for old result syntax once the spec tests
+    // are updated.
     if (s[i]->isStr()) {
-      // block signature
       curr->type = stringToWasmType(s[i++]->str());
-      if (i >= s.size()) break; // empty block
     } else {
-      curr->type = none;
+      Element& params = *s[i];
+      IString id = params[0]->str();
+      if (id == RESULT) {
+        curr->type = stringToWasmType(params[1]->str());
```

Comment:
This is just clunky enough and repeated enough to feel like it should be extracted to a function.
Something like
```c++
curr->type = getResultType(*s[i]);
if (curr->type != none) {
  i++;
}
```

Or if we want to be cute,
```c++
if ((curr->type = getResultType(*s[i])) != none) {
  i++;
}
```
(let's not be cute :p)